core/src/player.rs

```diff
@@ -1441,6 +1492,35 @@ impl Player {
         needs_render
     }
 
+    //Checks if two displayObjects have the same depth and id
+    fn check_display_object_equality(object1: DisplayObject, object2: DisplayObject) -> bool {
+        object1.depth() == object2.depth() && object1.id() == object2.id()
```

Comment:
This function should probably also check the movie, since everywhere this function is called, the movie is checked anyway.

Reply:
Almost certainly yes, and it'd also fix your other comment about `movie()` not being checked

Reply:
Thanks for the feedback, I am away for a few days. But will fix them when I'm back.

Reply:
Fixed those review comments and fixed some minor mistakes I made earlier. It can be reviewed again.